### PR TITLE
Upgrade ruff to 0.0.284

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Upgrade ruff to 0.0.284.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -369,8 +369,7 @@ def _get_value(cls, identifier_type, key, token):
                 return cls._parse_run_ids(token)
             else:
                 raise MlflowException(
-                    "Expected a quoted string value for attributes. "
-                    "Got value {value}".format(value=token.value),
+                    "Expected a quoted string value for attributes. " f"Got value {token.value}",
```
Comment (serena-ruan) [inline mlflow/utils/search_utils.py:372]:
```suggestion
                    "Expected a quoted string value for attributes. "
                    f"Got value {token.value}",
```

File: mlflow/models/wheeled_model.py [hunk 2]
```diff
@@ -222,8 +222,7 @@ def _download_wheels(cls, pip_requirements_path, dst_path):
             subprocess.run([download_command], check=True, shell=True, capture_output=True)
         except subprocess.CalledProcessError as e:
             raise MlflowException(
-                "An error occurred while downloading the dependency wheels: "
-                "{}".format(str(e.stderr))
+                "An error occurred while downloading the dependency wheels: " f"{str(e.stderr)}"
```
Comment (serena-ruan) [inline mlflow/models/wheeled_model.py:225]:
```suggestion
                "An error occurred while downloading the dependency wheels: "
                f"{str(e.stderr)}"
```

File: mlflow/store/model_registry/sqlalchemy_store.py [hunk 3]
```diff
@@ -209,8 +209,7 @@ def _get_registered_model(cls, session, name, eager=False):
             )
         if len(rms) > 1:
             raise MlflowException(
-                "Expected only 1 registered model with name={}. "
-                "Found {}.".format(name, len(rms)),
+                f"Expected only 1 registered model with name={name}. " f"Found {len(rms)}.",
```
Comment (serena-ruan) [inline mlflow/store/model_registry/sqlalchemy_store.py:212]:
```suggestion
                f"Expected only 1 registered model with name={name}. "
                f"Found {len(rms)}.",
```

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@serena-ruan Thanks for the review, fixed!

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**